contracts/Exchange.sol

```diff
@@ -154,22 +152,30 @@ contract Exchange is Restricted {
         return(activeBuyOrders.length, activeSellOrders.length);
     }
 
-    // returns CHUNK_SIZE orders starting from offset
+    // returns <chunkSize> active buy orders starting from <offset>
     // orders are encoded as [id, maker, price, amount]
-    function getActiveBuyOrders(uint offset) external view returns (uint[4][CHUNK_SIZE] response) {
-        for (uint8 i = 0; i < CHUNK_SIZE && i + offset < activeBuyOrders.length; i++) {
+    function getActiveBuyOrders(uint offset, uint16 chunkSize)
+    external view returns (uint[4][]) {
+        uint[4][] memory response = new uint[4][](chunkSize);
```

Comment:
The original reason for choosing this fixed `int` array was that the ABI decoders could not handle dynamic arrays, nor structs properly. It’s great that dynamic arrays are now supported. Is returning a struct an option? An `Order[]` would be ideal.

Reply:
Returning array of structs is currently an experimental feature (needs `pragma experimental ABIEncoderV2`). We'll have to wait a little for this change.